mitodl/micromasters
# Restored final grade histogram for selected courses in learner search

#### What are the relevant tickets?
Closes #2939 
Closes #3070 

#### What's this PR do?
Restores the final grade histogram for selected courses in learner search, and makes it work with the selected course.

#### How should this be manually tested?
Filter by a course in the learner search page, move the histogram bounds to filter by grade ranges, remove the course filter to make sure that the histogram also disappears, and test the histogram's interoperability with other available filters.

#### Where should the reviewer start?
`static/js/components/search/NestedAggregatingMenuFilter.js` , `static/js/components/search/FinalGradeRangeFilter.js`, `static/js/components/search/util.js`

#### Any background context you want to provide?
~~There's a lot of repeated code between the two files I mentioned above. I did my best to keep the implementation differences separate and obvious. The repetition is there because there's a somewhat complicated inheritance pattern. The accessors need to subclass from Searchkit classes in order to get some base functionality & properties, and they also share a lot of the same custom implementation that we've created. I'll open an issue to address this repetition *(issue added: #3070)*~~ The work described here is already merged into this PR

#### Screenshots (if appropriate)
<img width="997" alt="ss 2017-04-14 at 09 53 30" src="https://cloud.githubusercontent.com/assets/14932219/25045079/46b626b2-20f8-11e7-944c-87c73e3117db.png">
